Source/Core/Core/NetPlayClient.cpp

```diff
@@ -548,6 +563,216 @@ unsigned int NetPlayClient::OnData(sf::Packet& packet)
   }
   break;
 
+  case NP_MSG_SYNC_SAVE_DATA:
+  {
+    MessageId sub_id;
+    packet >> sub_id;
+
+    switch (sub_id)
+    {
+    case SYNC_SAVE_DATA_NOTIFY:
+    {
+      packet >> m_sync_save_data_count;
+      m_sync_save_data_success_count = 0;
+
+      if (m_sync_save_data_count == 0)
+        SyncSaveDataResponse(true);
+      else
+        m_dialog->AppendChat(GetStringT("Synchronizing save data..."));
+    }
+    break;
+
+    case SYNC_SAVE_DATA_RAW:
+    {
+      if (m_dialog->IsHosting())
+        return 0;
+
+      bool is_slot_a;
+      std::string region;
+      bool mc251;
+      packet >> is_slot_a >> region >> mc251;
+
+      const std::string path = File::GetUserPath(D_GCUSER_IDX) + GC_MEMCARD_NETPLAY +
+                               (is_slot_a ? "A." : "B.") + region + (mc251 ? ".251" : "") + ".raw";
+      if (File::Exists(path) && !File::Delete(path))
+      {
+        PanicAlertT("Failed to delete NetPlay memory card. Verify your write permissions.");
+        SyncSaveDataResponse(false);
+        return 0;
+      }
+
+      const bool success = DecompressPacketIntoFile(packet, path);
+      SyncSaveDataResponse(success);
+    }
+    break;
+
+    case SYNC_SAVE_DATA_GCI:
+    {
+      if (m_dialog->IsHosting())
+        return 0;
+
+      bool is_slot_a;
+      u8 file_count;
+      packet >> is_slot_a >> file_count;
+
+      const std::string path = File::GetUserPath(D_GCUSER_IDX) + GC_MEMCARD_NETPLAY DIR_SEP +
+                               StringFromFormat("Card %c", is_slot_a ? 'A' : 'B');
+
+      if ((File::Exists(path) && !File::DeleteDirRecursively(path + DIR_SEP)) ||
+          !File::CreateFullPath(path + DIR_SEP))
+      {
+        PanicAlertT("Failed to reset NetPlay GCI folder. Verify your write permissions.");
+        SyncSaveDataResponse(false);
+        return 0;
+      }
+
+      for (u8 i = 0; i < file_count; i++)
+      {
+        std::string file_name;
+        packet >> file_name;
+
+        if (!DecompressPacketIntoFile(packet, path + DIR_SEP + file_name))
+        {
+          SyncSaveDataResponse(false);
+          return 0;
+        }
+      }
+
+      SyncSaveDataResponse(true);
+    }
+    break;
+
+    case SYNC_SAVE_DATA_WII:
+    {
+      if (m_dialog->IsHosting())
+        return 0;
+
+      const auto game = m_dialog->FindGameFile(m_selected_game);
+      if (game == nullptr)
+      {
+        SyncSaveDataResponse(true);  // whatever, we won't be booting anyways
+        return 0;
+      }
+
+      const std::string path = File::GetUserPath(D_USER_IDX) + "Wii" GC_MEMCARD_NETPLAY DIR_SEP;
+
+      if (File::Exists(path) && !File::DeleteDirRecursively(path))
+      {
+        PanicAlertT("Failed to reset NetPlay NAND folder. Verify your write permissions.");
+        SyncSaveDataResponse(false);
+        return 0;
+      }
+
+      std::unique_ptr<IOS::HLE::FS::FileSystem> temp_fs =
```

Comment:
can't this just be `auto temp_fs = ...` or does the base class type actually matter here.

Reply:
Ah yeah, that's just leftover from a previous implementation.